xm-commons-cache/src/main/java/com/icthh/xm/commons/cache/service/DynamicCaffeineCacheManager.java

```diff
@@ -0,0 +1,140 @@
+package com.icthh.xm.commons.cache.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Ticker;
+import com.icthh.xm.commons.cache.TenantCacheManager;
+import com.icthh.xm.commons.cache.config.XmTenantLepCacheConfig;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.cache.caffeine.CaffeineCacheManager;
+import org.springframework.context.ApplicationListener;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.icthh.xm.commons.cache.config.XmTenantLepCacheConfig.CACHE_DEFAULTS;
+
+@RequiredArgsConstructor
+@Slf4j
+public class DynamicCaffeineCacheManager extends CaffeineCacheManager implements ApplicationListener<InitCachesEvent> {
+
+    private final ConcurrentMap<String, Supplier<Caffeine>> cacheCfgMap = new ConcurrentHashMap<>(16);
```

Comment:
In case we would use cacheCfgMap with nested tenant's map will it allow to avoid using of Lock?
Map<$tenant, Map<$cacheName, Supplier<Caffeine>>>